telegram/ext/updater.py

```diff
@@ -475,6 +527,23 @@ def bootstrap_clean_updates():
                 updates = self.bot.get_updates(updates[-1].update_id + 1)
             return False
 
+        def bootstrap_clean_updates_datetime(datetime_cutoff):
+            self.logger.debug('Cleaning updates from Telegram server with datetime "%s"',
+                                    datetime_cutoff)
+            updates = self.bot.get_updates()
+
+            # reversed as we just need to find the first msg that's too old
```

Comment:
Can we be sure that the updates are always ordered? I just don't know tbh.

Reply:
:facepalm: assumption is the mother of f..... i got a sequential list every time i ran my tests (and a ran a lot of m... *mumbling something about  dates and time*) 
I'll have look.

Reply:
- telegram bot API does not mention it is ordered
- i've looked at `get_updates()` but don't see ordering
- i've looked at `de_json` but don't see ordering

Note that the clean as bool part also assumes an ordered by update_id list.

we could do
`updates.sort(key=lambda x: x.update_id, reverse=True)`

maybe even better to put this in `get_updates()` before it returns the list

Reply:
Mh … Trusting TG to send updates with incremental `update_id` is okay. They implicity guarantee that by 
>In order to avoid getting duplicate updates, recalculate offset after each server response.
Trusting that the date necessarily increasing with the `update_id` is another thing. I don't see, why it shouldn't, but still … for the moment, just assume it is